docs/specs/component.md

```diff
@@ -103,101 +104,139 @@ events:
   individual events. For example, emitting the `EventsReceived` event for 10
   events MUST increment the `component_received_events_total` counter by 10.
 
+#### ComponentBilled
+
+*Sources* MUST emit a `ComponentBilled` event that repsents the billing of
+enterprise customers.
+
+- Emission
+  - MUST emit only if the `enterprise.enabled` config option is set to `true`.
+  - MUST emit immediatley before the creation of Vector events and after
+    decompression and filtering of raw bytes.
+- Properties
+  - `byte_size` - REQUIRED, number of raw bytes received that will be turned
+    into Vector events.
+- Metrics
+  - MUST increment the `component_billed_bytes_total` counter by the defined
+    value with the defined properties as metric tags.
+- Logs
+  - MUST log a `Bytes billed.` message at the `trace` level with the defined
+    properties as key-value pairs.
+  - MUST NOT be rate limited.
+
 #### ComponentBytesReceived
 
-*Sources* MUST emit a `ComponentBytesReceived` event immediately after receiving, decompressing
-and filtering bytes from the upstream source and before the creation of a Vector event.
+*Sources* MUST emit a `ComponentBytesReceived` event that represents the
+reception of *raw event bytes*. It MUST emit immediately after receiving
+bytes from the upstream source and before decompression, filtering, and the
+creation of Vector events.
```

Comment:
To clarify, you mean for this metric to indicate the total number of bytes received on, for example, a TCP socket? This would include HTTP headers, gRPC envelopes, etc.

Reply:
No, _not_ the envelope/framing, just the raw bytes used to create events. For HTTP, this would be the HTTP body only (AKA the value of `Content-Length`). For a socket, this would be the bytes received minus the framing/delimiter.

Reply:
I clarify this in the requirements below.